### ISSUE TITLE
In-Game: "_GMFILE_" returns "@@FILENAME@@" and "_GMFUNCTION_" returns "@@FUNC@@"

### ISSUE THREAD

**Issue by tinkerer-red** [YoYoGames/GameMaker-Bugs#7729]
### Description

our community tool `trace` no longer works as expected because `_GMFILE_` and `_GMFUNCTION_` are both returning `@@FILENAME@@` and `@@FUNC@@` respectively.

### Steps To Reproduce

```gml
#macro trace  __trace(_GMFILE_+"/"+_GMFUNCTION_+":"+string(_GMLINE_)+": ")
function __trace(_location) {
		static __struct = {};
		__struct.__location = _location;
		return method(__struct, function(_str)
    {
        show_debug_message(__location + ": " + string(_str));
    });
}
```

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.8.0 (Monthly)

### Which platform(s) are you seeing the problem on?

Windows

### Sample Package Attached?

- [X] I have attached my bug package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by DiasFranciscoA** [comment]
@tinkerer-red this only happens if you use `_GMFUNCTION_` or `_GMFILE_` inside a string interpolation or append it to a string otherwise it would work correctly. This is a bug still but you can still work around it by detailing the contatenation:

```
#macro trace  __trace(_GMFILE_, _GMFUNCTION_, _GMLINE_)
function __trace(_file, _func, _line) {
		static __struct = {};
		
		__struct.__location = $"{_file}:{_func}:{_line}";
		return method(__struct, function(_str)
    {
        show_debug_message(__location + ": " + string(_str));
    });
}
```

**Comment by rwkay** [comment]
Fixed in 2024.11

**Comment by scott-dunbar** [comment]
Verified fixed in  IDE v2024.1100.0.646  Runtime v2024.1100.0.666
